worker/src/direct_invocation/direct_invocation_ocall.rs

```diff
@@ -0,0 +1,152 @@
+/*
+    Copyright 2019 Supercomputing Systems AG
+    Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+use crate::direct_invocation::watch_list_service::WatchListFacade;
+use crate::ocall_bridge::bridge_api::{
+    DirectInvocationBridge, OCallBridgeError, OCallBridgeResult,
+};
+use codec::{Decode, Encode};
+use sp_core::H256 as Hash;
+use std::sync::Arc;
+use substratee_worker_primitives::{DirectRequestStatus, RpcReturnValue, TrustedOperationStatus};
+use ws::CloseCode;
+
+/// implementation of the direct invocation o-call
+pub struct DirectInvocationOCall<W> {
+    watch_list: Arc<W>,
```

Comment:
The watch list used to be a global variable/state, now it is shared state just for those components which need it, not global anymore.